main/src/ca/uwaterloo/flix/language/phase/Simplifier.scala

```diff
@@ -222,6 +239,35 @@ object Simplifier {
       throw InternalCompilerException(s"Unexpected expression: $exp0.", loc)
   }
 
+  /**
+   * Helper method to get a corresponding monotype to the given type.
+   */
+  private def getMonotype(tpe: Type): MonoType =
```

Comment:
Can we do `visitType` for this one?

Reply:
Oh it seems right. I removed the function and tried with visitType instead.